gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DatastoreFactory.java

```diff
@@ -16,18 +16,29 @@
 
 package com.google.gcloud.datastore;
 
+import com.google.gcloud.ServiceFactory;
+import java.io.ObjectStreamException;
 
 /**
  * A base class for Datastore factories.
  */
-public abstract class DatastoreFactory {
+public abstract class DatastoreFactory implements ServiceFactory<Datastore, DatastoreOptions> {
+
+  private static final long serialVersionUID = 5037190305022535983L;
 
   private static final DatastoreFactory INSTANCE = new DatastoreFactory() {
```

Comment:
@eamonnmcmanus I understand your concerns regarding serialization and the comment on serializing anonymous classes did not get lost. 

IMHO we should first focus on providing an easy to use, well designed library and then start thinking about more advanced usage that might involve serialization. Can we agree to put this PR "on hold" and go back to #55 (and other features) without adding more `Serializable`? We already have a "serializable" layer made of `BlobInfo` and `BucketInfo` that might be used by users that need it.